### ISSUE TITLE
Any plans to upgrade SetACL to the new version?

### ISSUE THREAD

**Issue by jpluimers** [bcurran3/ChocolateyPackages_CCR#309]
[SetACL 3.1.2](https://helgeklein.com/download/) has been available for quite a while now.

Any plans to upgrade to it from the current 3.0.6?

Are you open for pull request (since there are a few open ones)?

**Comment by bcurran3** [comment]
Thanks for letting me know. I’ll put it on the TDL.

 

Repo is out of sync at the moment, so I’m not pulling in anything until I have time to fix or normally I’d say yes (Though I think this is just changing the version number in the script.)

 

If this is something you care about and use, I’m open to adding you as a maintainer if you’re interested.

 

Bill

 

 

 

From: Jeroen Wiert Pluimers ***@***.***> 
Sent: Tuesday, October 4, 2022 12:13 PM
To: bcurran3/ChocolateyPackages ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [bcurran3/ChocolateyPackages] Any plans to upgrade SetACL to the new version? (Issue #309)

 

SetACL 3.1.2 <https://helgeklein.com/download/>  has been available for quite a while now.

Any plans to upgrade to it from the current 3.0.6?

Are you open for pull request (since there are a few open ones)?

**Comment by jpluimers** [comment]
Thanks for the maintainer state offer. I'm flattered, but decline: given the high chances of metastases returning and left-over symptoms after all the procedures I had, I'm actively cutting down on people/things/processes depending on me like (maintenance) responsibilities. Energy permitting I do provide pull requests and guidance.

The 3.0.6 version download file has disappeared, so now the package installation fails (see output below).

Today is a good day, so I will create a pull request for that.

For my reference:

1. Download link
   - old download link: https://helgeklein.com/downloads/SetACL/current/SetACL%20(executable%20version).zip (which now gives a 404-error)
   - new download link: https://helgeklein.com/downloads/SetACL/current/SetACL%203.1.2%20(executable%20version).zip
   - change in https://github.com/bcurran3/ChocolateyPackages/blob/master/setacl/tools/chocolateyInstall.ps1
2. SHA checksum
   - old sha256: `09e34e294c95ba7967c5463eca1acb571b745620cc2f58527dd1c1f154436ee0`
   - new sha256: `ba74399a70963c156580180455fbfc0fa68ea673a64eb89010a46273c7d478cc`
   - change for https://github.com/bcurran3/ChocolateyPackages/blob/master/setacl/tools/chocolateyInstall.ps1
3. Version number
   - old version number `3.0.6`
   - new version number `3.1.2`
   - quad position version number upgrade from `3.0.6.0` to `3.1.2.0` in https://github.com/bcurran3/ChocolateyPackages/blob/master/setacl/setacl.nuspec

Note that the [Chocolatey checksum validation](https://github.com/chocolatey/choco/blob/develop/src/chocolatey.resources/helpers/functions/Get-CheckSumValid.ps1) uses [`checksum.exe`](https://github.com/chocolatey/choco/blob/develop/src/chocolatey.resources/tools/checksum.exe) which compares hashes without case sensitivity. Since I'm from a [`certutil`](https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/certutil#-hashfile) and unix background that default to lowercase hashes for [test vectors](https://en.wikipedia.org/wiki/sha256#Test_vectors) the above sha256 hashes are lowercase.

I got to the [`checksum` source code](https://github.com/chocolatey/checksum) via [chocolatey/choco: Improve handling for checksum type #1018](https://github.com/chocolatey/choco/issues/1018)

Relevant source line: https://github.com/chocolatey/checksum/blob/master/src/checksum/Program.cs#L90

For historic purposes:
- the old version 3.0.6 download is still at https://web.archive.org/web/20181107043600/https://helgeklein.com/downloads/SetACL/current/SetACL%20(executable%20version).zip
- the new version 3.1.2 download is at https://web.archive.org/web/20220518094231/https://helgeklein.com/downloads/SetACL/current/SetACL%203.1.2%20(executable%20version).zip

Output:

``` text
C:\temp>choco upgrade --yes setacl
Chocolatey v1.1.0
Upgrading the following packages:
setacl
By upgrading, you accept licenses for the packages.
setacl is not installed. Installing...
Progress: Downloading setacl 3.0.6.0... 100%
Progress: Downloading setacl 3.0.6.0... 100%

setacl v3.0.6.0 [Approved]
setacl package files upgrade completed. Performing other installation steps.
Attempt to get headers for https://helgeklein.com/downloads/SetACL/current/SetACL%20(executable%20version).zip failed.
  The remote file either doesn't exist, is unauthorized, or is forbidden for url 'https://helgeklein.com/downloads/SetACL/current/SetACL%20(executable%20version).zip'. Exception calling "GetResponse" with "0" argument(s): "The remote server returned an error: (404) Not Found."
Downloading setacl
  from 'https://helgeklein.com/downloads/SetACL/current/SetACL%20(executable%20version).zip'
ERROR: The remote file either doesn't exist, is unauthorized, or is forbidden for url 'https://helgeklein.com/downloads/SetACL/current/SetACL%20(executable%20version).zip'. Exception calling "GetResponse" with "0" argument(s): "The remote server returned an error: (404) Not Found."
This package is likely not broken for licensed users - see https://docs.chocolatey.org/en-us/features/private-cdn.
The upgrade of setacl was NOT successful.
Error while running 'C:\ProgramData\chocolatey\lib\setacl\tools\chocolateyInstall.ps1'.
 See log for details.

Chocolatey upgraded 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - setacl (exited 404) - Error while running 'C:\ProgramData\chocolatey\lib\setacl\tools\chocolateyInstall.ps1'.
 See log for details.
```

**Comment by bcurran3** [comment]
Thanks. You put enough work into this that you should maintain it. :-)
Done.

**Comment by bcurran3** [comment]
p.s. Hope all is well and that you're procedures were all successful.

**Comment by jpluimers** [comment]
> p.s. Hope all is well and that you're procedures were all successful.

Yup. 18 months of hectic stuff behind us.

Long story short: I recovered from the surgery better than expected, we found a place for my mentally retarded brother to live internally with 24/7 care, did a big part of administration that had lagged behind for way too long, finally had time on our hands to think "now that we don't spend 20 hours care/week any more: what is good for ourselves to do?".

We decided that living in our split-level house would not be future proof, prepared the house for sale (basically by arranging some big storage units getting rid of excess stuff so they could make this video: https://youtu.be/uBMgRPrQdS0) and got ourselves a maisonette apartment with elevator which is also much closer to my brother. Similar view https://photos.app.goo.gl/xAVwnh347BUXNYkb8

In retrospect, the apartment had quite a few issues most of which are mostly solved by now (only 1/3 of rooms halfway unusable instead of only 1 room usable).

Which means I am becoming a bit more active on GitHub this year, awaiting more checks to see if the intestines are holding up nicely and no metastases evolving. The better half moved jobs and still is doing lots of physiotherapy to postpone the moment she gets into a wheelchair. She succeeded for 3 years now, so hopefully that too will be OK enough for a couple of years.

Glad the change request got merged last year, I can now finally ditch the testing VM I had for it (: 

--jeroen
